crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -135,11 +136,28 @@ impl Extension for QueryLimitsChecker {
         variables: &Variables,
         next: NextParseQuery<'_>,
     ) -> ServerResult<ExecutableDocument> {
+        let latency_timer = ctx
+            .data_opt::<Metrics>()
+            .map(|x| x.request_metrics.query_validation_latency.start_timer());
+        let timer_latency_by_path = ctx.data_opt::<Metrics>().map(|x| {
+            x.request_metrics
+                .query_validation_latency_by_path
+                // TODO make the path here
+                .with_label_values(&["path"])
+                .start_timer()
+        });
         let cfg = ctx
             .data::<ServiceConfig>()
             .expect("No service config provided in schema data");
 
         if query.len() > cfg.limits.max_query_payload_size as usize {
+            if let Some(metrics) = ctx.data_opt::<Arc<Metrics>>() {
+                metrics
+                    .request_metrics
+                    .query_payload_error
+                    .with_label_values(&["validation"])
```

Comment:
the label does not look right, we probably don't want a label here.

Reply:
Me noob: what do labels do?

Reply:
Labels are a way to group values by some key. E.g., you could use a status label, and then have the values 200, 400, 500, each with its related values.